ocs_ci/ocs/ocs_upgrade.py

```
@@ -0,0 +1,359 @@
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_RBDPLUGIN_PROVISIONER_LABEL,
+        count=2
+    )
+    verify_pods_upgraded(old_images, selector=constants.MON_APP_LABEL, count=3)
+    verify_pods_upgraded(old_images, selector=constants.MGR_APP_LABEL)
+    verify_pods_upgraded(
+        old_images, selector=constants.OSD_APP_LABEL, count=osd_count,
+        timeout=1800,
+    )
+    verify_pods_upgraded(old_images, selector=constants.MDS_APP_LABEL, count=2)
+    if config.ENV_DATA.get('platform') == constants.VSPHERE_PLATFORM:
+        verify_pods_upgraded(
+            old_images, selector=constants.RGW_APP_LABEL, count=1
+        )
+
+
+class OCSUpgrade(object):
+    """
+    OCS Upgrade helper class
+
+    """
+
+    namespace = config.ENV_DATA['cluster_namespace']
+    version_before_upgrade = config.ENV_DATA.get("ocs_version")
+    ocs_registry_image = config.UPGRADE.get('upgrade_ocs_registry_image')
+    parsed_version_before_upgrade = parse_version(version_before_upgrade)
+    upgrade_in_current_source = config.UPGRADE.get(
+        'upgrade_in_current_source', False
+    )
+    upgrade_version = config.UPGRADE.get(
+        "upgrade_ocs_version", version_before_upgrade
+    )
+    if ocs_registry_image:
+        upgrade_version = get_ocs_version_from_image(
+            ocs_registry_image
+        )
+    parsed_upgrade_version = parse_version(upgrade_version)
+
+    def load_version_config_file(self, upgrade_version):
+        """
+        Loads config file to the ocs-ci config with upgrade version
+
+        Args:
+            upgrade_version: (str): version to be upgraded
+
+        """
+
+        version_change = self.parsed_upgrade_version > self.parsed_version_before_upgrade
+        if version_change:
+            version_config_file = os.path.join(
+                constants.CONF_DIR, 'ocs_version', f'ocs-{upgrade_version}.yaml'
+            )
+            load_config_file(version_config_file)
```

what happens else? Should we throw an exception or at least log a warning?

---

will add a warning

---

This is expected as if it's upgrade between Z stream the X.Y version will not change hence no need to reload config for specific version like between 4.3 and 4.4 . So not sure if it should be warning level as nothing wrong when upgrade from 4.4.0 and 4.4.1 for example as X.Y version will not change.